src/pds/registrysweepers/reindexer/main.py

```diff
@@ -0,0 +1,238 @@
+import logging
+from datetime import datetime
+from datetime import timezone
+from typing import Dict
+from typing import Iterable
+from typing import Union
+
+from opensearchpy import OpenSearch
+from pds.registrysweepers.reindexer.constants import REINDEXER_FLAG_METADATA_KEY
+from pds.registrysweepers.utils import configure_logging
+from pds.registrysweepers.utils import parse_args
+from pds.registrysweepers.utils.db import query_registry_db_with_search_after
+from pds.registrysweepers.utils.db import write_updated_docs
+from pds.registrysweepers.utils.db.client import get_userpass_opensearch_client
+from pds.registrysweepers.utils.db.indexing import ensure_index_mapping
+from pds.registrysweepers.utils.db.multitenancy import resolve_multitenant_index_name
+from pds.registrysweepers.utils.db.update import Update
+
+log = logging.getLogger(__name__)
+
+
+def get_docs_query(filter_to_harvested_before: datetime):
+    """
+    Return a query to get all docs which haven't been reindexed by this sweeper and which haven't been harvested
+    since this sweeper process instance started running
+    """
+    # TODO: Remove this once query_registry_db_with_search_after is modified to remove mutation side-effects
```

Comment:
@tloubrieu-jpl sorry I missed these comments (if i did? They're stamped Oct 14)

Regarding the docstring, I'll word it more clearly and update the PR.

- Get all docs
- Exclude anything which has already been processed by the sweeper
- Exclude anything which was harvested in the middle of a reindexer sweeper run, as picking them up would cause erroneous behaviour.

Regarding the `TODO`, query_registry_db_with_search_after() has a side-effect of mutating the query dict it's passed.

This is bad and needs to be fixed, because if you define a query as a variable so it can be reused (as it is in this sweeper), the second time you pass it it will have erroneous search-after metadata attached to it, causing a fatal error.

The function you commented on is workaround for this bug, since fixing it is non-trivial and needs a little thought.